core/shared/src/main/scala-2.13+/zio/BuildFromCompat.scala

```diff
@@ -6,7 +6,12 @@ private[zio] trait BuildFromCompat {
 
   type BuildFrom[-From, -A, +C] = scala.collection.BuildFrom[From, A, C]
 
-  implicit def buildFromAny[Element, Collection[+Element] <: Iterable[Element] with IterableOps[Any, Collection, Any]]
+  @deprecated("Use BuildFrom.buildFromIterableOps or buildFromNothing instead", "1.0.6")
```

Comment:
If we're going to change this can we just delete it entirely instead of deprecating it?

Reply:
Wouldn't that be binary incompatible?

Reply:
Binary compatibility checks seem to be okay.

Reply:
Oh no - it loos like the MiMa checks job has been broken for some time 😱 
https://circleci.com/api/v1.1/project/github/zio/zio/107145/output/104/0?file=true&allocation-id=605246e5efc352258c281098-0-build%2F701E1D91

Reply:
Or was that on purpose?

Reply:
Ok I see
```
  // No bincompat on streams yet
  // No bincompat on zio-test yet
```

Reply:
No. We don't guarantee binary compatibility on ZIO Stream or ZIO Test and the other items were specifically excluded.

Reply:
@adamgfraser oh I see why - we check binary compatibility against 1.0.1 but this was added in 1.0.4-1: https://github.com/zio/zio/releases/tag/v1.0.4-1

Wow, I never really thought about this, but if you don't bump the version for MiMa that means you could accidentally delete something that you introduced in a later version. Amazing 🤯

Reply:
Should I bump the version in `MimaSettings`?

Reply:
Yes we should.

Reply:
That makes me think there should be some process or automation to bump this version on release

Reply:
Yes. Not sure what the best way is to do it but that would definitely be good.